dev-java/protobuf-java/protobuf-java-3.20.1.ebuild

```diff
@@ -0,0 +1,90 @@
+# Copyright 2008-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# Tests not enabled, depend on com.google.truth which is not packaged
+# https://github.com/protocolbuffers/protobuf/blob/v21.4/java/core/pom.xml#L35-L40
+JAVA_PKG_IUSE="doc source"
+MAVEN_ID="com.google.protobuf:${PN}:${PV}"
+
+inherit java-pkg-2 java-pkg-simple
+
+PARENT_PN="${PN/-java/}"
+PARENT_PV="${PV}"
+PARENT_P="${PARENT_PN}-${PARENT_PV}"
+
+if [[ "${PV}" == "9999" ]]; then
+	inherit git-r3
+
+	EGIT_REPO_URI="https://github.com/protocolbuffers/${PARENT_PN}"
+	EGIT_SUBMODULES=()
+	EGIT_CHECKOUT_DIR="${WORKDIR}/${PARENT_P}"
+else
+	inherit verify-sig
+	SRC_URI="
+		https://github.com/protocolbuffers/${PARENT_PN}/archive/v${PARENT_PV}.tar.gz
+			-> ${PARENT_P}.tar.gz
+	"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86 ~amd64-linux ~x86-linux ~x64-macos"
+fi
+
+DESCRIPTION="Google's Protocol Buffers - Java bindings"
+HOMEPAGE="
+	https://developers.google.com/protocol-buffers/
+	https://github.com/protocolbuffers/protobuf
+"
+
+LICENSE="BSD"
+SLOT="0/31"
+
+S="${WORKDIR}/${PARENT_P}/java"
+
+BDEPEND="
+	dev-libs/${PARENT_PN}:${SLOT}
+	>=virtual/jdk-1.8:*
+"
+
+DEPEND=">=virtual/jdk-1.8:*"
+RDEPEND=">=virtual/jre-1.8:*"
+
+JAVA_SRC_DIR="core/src/main/java"
+JAVA_RESOURCE_DIRS="core/src/main/resources"
+JAVA_AUTOMATIC_MODULE_NAME="com.google.protobuf"
+JAVA_JAR_FILENAME="protobuf.jar"
+
+src_prepare() {
+	pushd "${WORKDIR}/${PARENT_P}" > /dev/null || die
+	default
+	popd > /dev/null || die
```

Comment:
Are thjose three lines really needed?

Reply:
Yes if we consider providing patches from the root of the project, like you can see in the Python and Ruby parts.
Otherwise your patches will need to start from the `java` directory.

Reply:
> Yes if we consider providing patches from the root of the project, like you can see in the Python and Ruby parts.
> Otherwise your patches will need to start from the `java` directory.

I'd add those lines only once needed, if ever.  Regarding [upstream's release frequency](https://mvnrepository.com/artifact/com.google.protobuf/protobuf-java) I doubt upstream patches are needed in Gentoo.

Reply:
It doesn’t hurt to keep those instructions, like it was done in Python and Ruby.
I now recall I forgot to provide the array `PATCH` variable, currently empty.